dataframe-csv/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/DelimCsvTsvTests.kt

```diff
@@ -0,0 +1,657 @@
+        val dfHeader = DataFrame.readCsv(simpleCsv, readLines = 0)
+        dfHeader.rowsCount() shouldBe 0
+        dfHeader.columnNames() shouldBe expected
+
+        val dfThree = DataFrame.readCsv(simpleCsv, readLines = 3)
+        dfThree.rowsCount() shouldBe 3
+
+        val dfFull = DataFrame.readCsv(simpleCsv, readLines = 10)
+        dfFull.rowsCount() shouldBe 5
+    }
+
+    @Test
+    fun `if string starts with a number, it should be parsed as a string anyway`() {
+        val df = DataFrame.readCsv(durationCsv)
+        df["duration"].type() shouldBe typeOf<String>()
+        df["floatDuration"].type() shouldBe typeOf<String>()
+    }
+
+    @Test
+    fun `if record has fewer columns than header then pad it with nulls`() {
+        @Language("CSV")
+        val csvContent =
+            """
+            col1,col2,col3
+            568,801,587
+            780,588
+            """.trimIndent()
+
+        val df = shouldNotThrowAny {
+            DataFrame.readCsvStr(csvContent)
+        }
+
+        df shouldBe dataFrameOf("col1", "col2", "col3")(
+            568, 801, 587,
+            780, 588, null,
+        )
+    }
+
+    @Test
+    fun `write and read frame column`() {
+        val df = dataFrameOf("a", "b", "c")(
+            1, 2, 3,
+            1, 3, 2,
+            2, 1, 3,
+        )
+        val grouped = df.groupBy("a").into("g")
+        val str = grouped.toCsvStr(escapeChar = null)
+        val res = DataFrame.readCsvStr(str, quote = '"')
+        res shouldBe grouped
+    }
+
+    @Test
+    fun `write and read column group`() {
+        val df = dataFrameOf("a", "b", "c")(
+            1, 2, 3,
+            1, 3, 2,
+        )
+        val grouped = df.group("b", "c").into("d")
+        val str = grouped.toCsvStr()
+        val res = DataFrame.readCsvStr(str)
+        res shouldBe grouped
+    }
+
+    @Test
+    fun `CSV String of saved dataframe starts with column name`() {
+        val df = dataFrameOf("a")(1)
+        df.toCsvStr().first() shouldBe 'a'
+    }
+
+    @Test
+    fun `guess tsv`() {
+        val df = DataFrame.read(testResource("abc.tsv"))
+        df.columnsCount() shouldBe 3
+        df.rowsCount() shouldBe 2
+    }
+
+    @Test
+    fun `write csv without header produce correct file`() {
+        val df = dataFrameOf("a", "b", "c")(
+            1, 2, 3,
+            1, 3, 2,
+        )
+        df.writeCsv(
+            path = "src/test/resources/without_header.csv",
+            includeHeader = false,
+            recordSeparator = "\r\n",
+        )
+        val producedFile = File("src/test/resources/without_header.csv")
+        producedFile.exists() shouldBe true
+        producedFile.readText() shouldBe "1,2,3\r\n1,3,2\r\n"
+        producedFile.delete()
+    }
+
+    @Test
+    fun `check integrity of example data`() {
+        val df = DataFrame.readCsv(
+            "../data/jetbrains_repositories.csv",
+            skipLines = 1, // now needs this, ignoreEmptyLines cannot catch it
```

Comment:
let's update the data?

Reply:
I think it's good that we know the difference in behavior for the new implementation; what happens if we try to read a csv file with a blank line at the top.